standards/application/messaging-api.md

```diff
@@ -0,0 +1,965 @@
+Content-Type: application/json
+
+{
+  "requestId": "9978c4f1-4465-4537-9d95-0cf016fb1080"
+}
+```
+
+Response:
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "status": "ok"
+}
+```
+
+##### `getPendingRequests`
+Request:
+```http
+GET /send/requests HTTP/1.1
+Accept: application/json
+```
+
+Response:
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+[
+  "9978c4f1-4465-4537-9d95-0cf016fb1080"
+]
+```
+
+### Subscribe
+
+The `Subscribe API` SHOULD be used to initiate a subscription over [FILTER](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md) or [RELAY](https://github.com/vacp2p/rfc-index/blob/0277fd0c4dbd907dfb2f0c28b6cde94a335e1fae/waku/standards/core/11/relay.md) for continuous message retrieval.
+
+When the `Subscribe API` is invoked, the following behaviors MUST occur.
+
+A subscription is established via [FILTER](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md) or [RELAY](https://github.com/vacp2p/rfc-index/blob/0277fd0c4dbd907dfb2f0c28b6cde94a335e1fae/waku/standards/core/11/relay.md) based on the node’s configuration for ongoing message reception.
+
+If `confirmStored` is set to `true`,
+a recurring [STORE](../standards/core/store.md) query MUST be initiated for the specified `contentTopics` to continuously update the stored property in the `MessageRecord` (as defined in the `Message Storage API`) for messages circulating in the network after the `Subscribe API` call.
+For details on the recurring query, refer to the `Message Storage API` section.
+
+Once the `Subscribe API` is called, the received property in the `MessageRecord` (as described in the `Message Storage API`) MUST be populated.
+
+When the subscription is active, the API MUST trigger the `message:*` event group as described in the `Event Source API` section.
+
+#### Programmatic API
+
+```typescript
+type SubscriptionID = string;
+
+type SubscriptionListener = (message: MessageRecord) => void | Promise<void>;
+
+interface ISubscribe {
+  public subscribe(decoder: Decoder, fn: SubscriptionListener): SubscriptionID;
+  public unsubscribe(id: SubscriptionID): void;
+  public getActiveSubscriptions(): SubscriptionID[];
+}
+```
+
+##### `SubscriptionID`
+A unique identifier (GUID string) representing a subscription.
+This ID MUST be used to manage active subscriptions,
+allowing them to be terminated when they are no longer needed.
+
+##### `SubscriptionListener`
+A callback function that is invoked whenever a new `MessageRecord` is received.
+It processes the incoming message and MAY return a `Promise` to support asynchronous handling.
+The `MessageRecord` is defined in the `Message Storage API` section.
+
+##### `subscribe`
+Registers a new subscription to listen for messages processed by a given `Decoder`.
+The subscription is based on the `pubsubTopic` and `contentTopic` provided by the `Decoder`,
```

Comment:
As I mentioned above - would it make sense to move away from `pubsubTopic` in dev/user facing parts and only refer to `shard`?